Use system-openssl in Makefile.
Use /usr/bin/wish in gitk.
Fix perl install locations.
Point git -w to %p/share/doc/git.

diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -215,10 +215,10 @@
 bindir = $(prefix)/$(bindir_relative)
 mandir = share/man
 infodir = share/info
-gitexecdir = libexec/git-core
+gitexecdir = lib/git
 sharedir = $(prefix)/share
 template_dir = share/git-core/templates
-htmldir = share/doc/git-doc
+htmldir = share/doc/git
 ifeq ($(prefix),/usr)
 sysconfdir = /etc
 ETC_GITCONFIG = $(sysconfdir)/gitconfig
@@ -231,7 +231,7 @@
 
 # default configuration for gitweb
 GITWEB_CONFIG = gitweb_config.perl
-GITWEB_CONFIG_SYSTEM = /etc/gitweb.conf
+GITWEB_CONFIG_SYSTEM = $(sysconfdir)/gitweb.conf
 GITWEB_HOME_LINK_STR = projects
 GITWEB_SITENAME =
 GITWEB_PROJECTROOT = /pub/git
@@ -857,6 +857,8 @@
 -include config.mak
 
 ifeq ($(uname_S),Darwin)
+	BASIC_CFLAGS += -I%p/lib/system-openssl/include -I%p/include
+	BASIC_LDFLAGS += -L%p/lib/system-openssl/lib -L%p/lib
 	ifndef NO_FINK
 		ifeq ($(shell test -d /sw/lib && echo y),y)
 			BASIC_CFLAGS += -I/sw/include
diff --git a/perl/Makefile b/perl/Makefile
--- a/perl/Makefile
+++ b/perl/Makefile
@@ -38,7 +38,7 @@
 	echo '	echo $(instdir_SQ)' >> $@
 else
 $(makfile): Makefile.PL ../GIT-CFLAGS
-	$(PERL_PATH) $< PREFIX='$(prefix_SQ)'
+	$(PERL_PATH) $< PREFIX='$(prefix_SQ)' LIB='%p/lib/perl5' INSTALLSITEMAN3DIR='%p/share/man/man3'
 endif
 
 # this is just added comfort for calling make directly in perl dir
